### ISSUE TITLE
Validation errors and alerts for the Concept Page

### ISSUE THREAD

**Issue by miguelvaara** [NatLibFi/Skosmos#1042]
Check up by visiting http://finto.fi/yso/en/page/p7972?clang=fi

Before fixing the errors mentioned in this issue make sure the issues #1036, #1037, #1038 and #1039 are done. They are more context oriented and this issue can only be fixed if the context is reasonable (semantic structure, header levels and tab order)

* Errors:

`- Missing form labels:`

Case 1:
`<input style="display: none" name="clang" value="fi" id="lang-input">`

Case 2:
`<input name="anylang" type="checkbox">`

Case 3:
`<input type="radio" id="all-languages-true" name="anylang" value="on" onclick="createCookie('SKOSMOS_SEARCH_ALL', this.value , 365);">`

A form control does not have a corresponding label.

What To Do:
If a text label for a form control is visible, use the <label> element to associate it with its respective form control. If there is no visible label, either provide an associated label, add a descriptive title attribute to the form control, or reference the label(s) using aria-labelledby. Labels are not required for image, submit, reset, button, or hidden form controls.

`- Empty button:`

Case 1:
```
<button type="button" data-toggle="tooltip" data-placement="button" title="Copy to clipboard" class="btn btn-default btn-xs copy-clipboard" for="#pref-label">
ARIA hidden<span class="glyphicon glyphicon-copy" aria-hidden="true"></span>
</button>
```

Case 2:
```
<button type="button" data-toggle="tooltip" data-placement="button" title="Copy to clipboard" class="btn btn-default btn-xs copy-clipboard" for="#uri-input-box">
ARIA hidden<span class="glyphicon glyphicon-copy" aria-hidden="true"></span>
</button>
```

What It Means
A button is empty or has no value text.

What To Do
Place text content within the <button> element or give the <input> element a value attribute.

* Alerts:

`- Orphaned form label:`

Case 1:
```
<label class="sr-only" for="lang-dropdown-toggle">
Content and search language
</label>
```

Case 2:
```
<label class="sr-only" for="search-all-button">
Submit search
</label>
```

A form label is present, but it is not correctly associated with a form control.

What To Do:
Properly associate the label with its corresponding form control. If there is no corresponding form control, remove the label. Labels are not appropriate for image, submit, reset, button, or hidden form controls.

`- Missing fieldset:`

Case 1:
`<input name="anylang" type="checkbox">`

What It Means
A group of check boxes or radio buttons is not enclosed in a fieldset.

What To Do
Determine whether the grouping of check boxes or radio buttons has or needs text that explains the purpose of the check boxes or radio button grouping. If so, mark up the group within a fieldset and put the group description in a legend element.

`- Redundant link:`

Case 1:
```
<a class="propertyvalue hidden-breadcrumb" href="yso/en/page/p8691?clang=fi">
ominaisuudet
</a>
```

Case 2:
```
<a href="yso/en/page/p1310?clang=fi">
energia
</a>
```

What It Means
Adjacent links go to the same URL.
reasonable
What To Do
If possible, combine the redundant links into one link and remove any redundant text or alternative text (for example, if a product image and product name are in the same link, the image can usually be given alt="").

**Comment by osma** [comment]
This is relevant for Skosmos 2 only.

We will make a new accessibility review for Skosmos 3, see #1871

Closing this issue as it is no longer relevant for current development.
